assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js

```diff
@@ -42,93 +51,146 @@ import BannerGraphicsSVGDesktop from '../../../../../../svg/graphics/audience-se
 import BannerGraphicsSVGTablet from '../../../../../../svg/graphics/audience-segmentation-setup-tablet.svg';
 import BannerGraphicsSVGMobile from '../../../../../../svg/graphics/audience-segmentation-setup-mobile.svg';
 
-function AudienceSegmentationSetupCTAWidget( { Widget, title, description } ) {
+const { useSelect, useDispatch } = Data;
+
+function AudienceSegmentationSetupCTAWidget( { Widget } ) {
 	const [ isSaving, setIsSaving ] = useState( false );
 	const breakpoint = useBreakpoint();
 	const isMobileBreakpoint = breakpoint === BREAKPOINT_SMALL;
 	const isTabletBreakpoint = breakpoint === BREAKPOINT_TABLET;
 
-	const onEnableGroups = useCallback( () => {
+	const { enableAudienceGroup } = useDispatch( MODULES_ANALYTICS_4 );
+
+	const onEnableGroups = useCallback( async () => {
 		setIsSaving( true );
-	}, [] );
 
+		await enableAudienceGroup();
+	}, [ enableAudienceGroup ] );
+
+	const configuredAudiences = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getConfiguredAudiences()
+	);
+
+	const hasDataWithinPast90Days = useSelect( ( select ) => {
+		const endDate = select( CORE_USER ).getReferenceDate();
+
+		const startDate = getPreviousDate(
+			endDate,
+			90 + DATE_RANGE_OFFSET // Add offset to ensure we have data for the entirety of the last 90 days.
+		);
+
+		const args = {
+			metrics: [ { name: 'totalUsers' } ],
+			startDate,
+			endDate,
+		};
+
+		return select( MODULES_ANALYTICS_4 ).hasZeroData( args ) === false;
+	} );
+
+	if ( configuredAudiences !== null || ! hasDataWithinPast90Days ) {
```

Comment:
Hey @hussain-t, just want to flag I made an additional change following your review which is to ensure the banner doesn't display while `configuredAudiences` is loading and therefore `undefined`. The change is in this commit: https://github.com/google/site-kit-wp/pull/8694/commits/51540e5ea89163f3df4b62b92d503fd937853ef6

Reply:
Sounds good. Thanks, @techanvil!